<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>774</width>
    <height>571</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout" stretch="1,12">
    <item>
     <widget class="QLabel" name="fileNameLabel">
      <property name="frameShape">
       <enum>QFrame::Panel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>4</number>
      </property>
      <property name="text">
       <string>Current File: None</string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,1">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>Configure Inputs</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2" stretch="9,1">
         <item>
          <widget class="InputList" name="inputList">
           <property name="layoutDirection">
            <enum>Qt::LeftToRight</enum>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOn</enum>
           </property>
           <property name="itemAlignment">
            <set>Qt::AlignLeading</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="actionNewInput">
             <property name="text">
              <string>New Input</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="actionDeleteSelectedInput">
             <property name="text">
              <string>Delete Selected</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>Input Macro</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QScrollArea" name="scrollArea_2">
           <property name="widgetResizable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="scrollAreaWidgetContents_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>345</width>
              <height>405</height>
             </rect>
            </property>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>774</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="PresetMenu" name="menuPresets">
    <property name="title">
     <string>Presets</string>
    </property>
    <addaction name="actionCreateNewPreset"/>
    <addaction name="actionLoadPreset"/>
    <addaction name="actionSavePresetAuto"/>
    <addaction name="actionSavePreset"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout_OpenMacro_GUI"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
   </widget>
   <widget class="DebugMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="actionTest_Serial_Write"/>
    <addaction name="actionTest_Serial_Read"/>
   </widget>
   <addaction name="menuPresets"/>
   <addaction name="menuSettings"/>
   <addaction name="menuHelp"/>
   <addaction name="menuDebug"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionAbout_OpenMacro_GUI">
   <property name="text">
    <string>Manual</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionLoadPreset">
   <property name="text">
    <string>Load Preset</string>
   </property>
  </action>
  <action name="actionSavePreset">
   <property name="text">
    <string>Save Preset As</string>
   </property>
  </action>
  <action name="actionTest_Serial_Write">
   <property name="text">
    <string>Test Serial Write</string>
   </property>
  </action>
  <action name="actionTest_Serial_Read">
   <property name="text">
    <string>Test Serial Read</string>
   </property>
  </action>
  <action name="actionCreateNewPreset">
   <property name="text">
    <string>Create New Preset</string>
   </property>
  </action>
  <action name="actionSavePresetAuto">
   <property name="text">
    <string>Save Preset</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DebugMenu</class>
   <extends>QMenu</extends>
   <header>App/debugmenu.h</header>
   <slots>
    <slot>onTestSerialWrite()</slot>
    <slot>onTestSerialRead()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>PresetMenu</class>
   <extends>QMenu</extends>
   <header>App/presetmenu.h</header>
   <slots>
    <slot>onCreateNewPreset()</slot>
    <slot>onLoadPreset()</slot>
    <slot>onSavePreset()</slot>
    <slot>onSavePresetAuto()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>InputList</class>
   <extends>QListWidget</extends>
   <header>App/inputlist.h</header>
   <slots>
    <slot>onNewInput()</slot>
    <slot>onDeleteSelected()</slot>
    <slot>onSelectChanged()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionNewInput</sender>
   <signal>clicked()</signal>
   <receiver>inputList</receiver>
   <slot>onNewInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>501</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDeleteSelectedInput</sender>
   <signal>clicked()</signal>
   <receiver>inputList</receiver>
   <slot>onDeleteSelected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>286</x>
     <y>501</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>inputList</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>inputList</receiver>
   <slot>onSelectChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>197</x>
     <y>294</y>
    </hint>
    <hint type="destinationlabel">
     <x>197</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTest_Serial_Read</sender>
   <signal>triggered()</signal>
   <receiver>menuDebug</receiver>
   <slot>onTestSerialRead()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>84</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTest_Serial_Write</sender>
   <signal>triggered()</signal>
   <receiver>menuDebug</receiver>
   <slot>onTestSerialWrite()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>84</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCreateNewPreset</sender>
   <signal>triggered()</signal>
   <receiver>menuPresets</receiver>
   <slot>onCreateNewPreset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoadPreset</sender>
   <signal>triggered()</signal>
   <receiver>menuPresets</receiver>
   <slot>onLoadPreset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSavePreset</sender>
   <signal>triggered()</signal>
   <receiver>menuPresets</receiver>
   <slot>onSavePreset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSavePresetAuto</sender>
   <signal>triggered()</signal>
   <receiver>menuPresets</receiver>
   <slot>onSavePresetAuto()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>Test()</slot>
 </slots>
</ui>
